### PR TITLE
Feature/firebase emulator

### PR DESCRIPTION
**Pull request by kiledal**

### CODE REVIEW

File: src/mock/index.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+const url = `http://localhost:${PORT}/${PROJECT_ID}/${REGION}/${FUNCTION_ID}?secret=${SECRET}`;
+
+run();
+
+async function run() {
+  let users;
+
+  try {
+    const systemUsers = require('./users/systemUsers.json');
+    const customUsers = require('./users/customUsers.json');
```
Comment (aulonm) [inline src/mock/index.js:26]: the actual file is named customUsers-template.json and not customUsers.json

File: src/views/Admin/AdminHome.vue [hunk 2]
```diff
@@ -5,8 +5,10 @@
     <div class="nav-wrapper">
       <div class="container">
         <nav class="nav">
-          <router-link class="router-link" :to="{ name: 'admin-users' }">{{ $t('admin.home.users') }}</router-link>
-          <router-link class="router-link" :to="{ name: 'admin-objects' }">{{
+          <router-link id="admin-users" class="router-link" :to="{ name: 'admin-users' }">{{
```
Comment (aulonm) [inline src/views/Admin/AdminHome.vue:8]: should maybe be data-cy?

File: src/views/Admin/AdminObjects.vue [hunk 3]
```diff
@@ -43,7 +43,12 @@
             {{ dept.name }}
             <i v-if="dept.archived" class="fa fas fa-history"></i>
           </div>
-          <div v-if="selectedOrgId" class="miller__col__item miller__col__add" @click="addDepartment">
+          <div
+            v-if="selectedOrgId"
+            id="admin-addDepartment"
+            class="miller__col__item miller__col__add"
+            @click="addDepartment"
+          >
```
Comment (aulonm) [inline src/views/Admin/AdminObjects.vue:51]: data-cy and not id

File: src/views/Admin/AdminObjects.vue [hunk 4]
```diff
@@ -61,7 +66,12 @@
 
             <i v-if="product.archived" class="fa fas fa-history"></i>
           </div>
-          <div v-if="selectedDeptId" class="miller__col__item miller__col__add" @click="addProduct">
+          <div
+            v-if="selectedDeptId"
+            id="admin-addProduct"
+            class="miller__col__item miller__col__add"
+            @click="addProduct"
+          >
```
Comment (aulonm) [inline src/views/Admin/AdminObjects.vue:74]: data-cy and not id

File: src/views/Admin/components/AdminDepartment.vue [hunk 5]
```diff
@@ -2,11 +2,13 @@
...
     <h2 class="title title-2">{{ $t('admin.department.title') }}</h2>
 
+    <CalloutArchivedRestore v-if="department.archived" :docref="docref"></CalloutArchivedRestore>
+
     <div class="section">
       <div class="form-group">
         <label class="form-field">
           <span class="form-label">{{ $t('admin.department.name') }}</span>
-          <input type="text" v-model="department.name" @input="updateSlug" maxlength="64" />
+          <input type="text" id="dep-name" v-model="department.name" @input="updateSlug" maxlength="64" />
```
Comment (aulonm) [inline src/views/Admin/components/AdminDepartment.vue:11]: data-cy and not id

File: src/views/Admin/components/AdminDepartment.vue [hunk 6]
```diff
@@ -34,13 +36,21 @@
...
           <span class="form-label">{{ $t('admin.department.missionStatement') }}</span>
           <span class="form-help" v-html="$t('admin.department.missionStatementHelp')"></span>
-          <textarea rows="4" v-model="department.missionStatement" @input="dirty = true" maxlength="320"></textarea>
+          <textarea
+            id="dep-missionStatement"
+            rows="4"
+            v-model="department.missionStatement"
+            @input="dirty = true"
+            maxlength="320"
+          ></textarea>
```
Comment (aulonm) [inline src/views/Admin/components/AdminDepartment.vue:45]: data-cy and not id?

File: src/views/Admin/components/AdminDepartment.vue [hunk 7]
```diff
@@ -34,13 +36,21 @@
...
       </div>
     </div>
 
-    <button class="btn" :disabled="!dirty" @click="saveObject">{{ $t('btn.save') }}</button>
-    <button class="btn btn--borderless" @click="deleteObject">{{ $t('admin.department.delete') }}</button>
+    <button class="btn" id="btn-saveDep" :disabled="!dirty" @click="saveObject">{{ $t('btn.save') }}</button>
+    <button class="btn btn--borderless" id="btn-deleteDep" @click="deleteObject">
+      {{ $t('admin.department.delete') }}
+    </button>
   </div>
```
Comment (aulonm) [inline src/views/Admin/components/AdminDepartment.vue:54]: data-cy and not id?

File: tests/e2e/support/createProduct.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { testDepartment, testProducts } from '../config';
+
+Cypress.Commands.add('createProduct', productIndex => {
+  cy.visit('/admin').wait(2000);
+
+  cy.get('[id="admin-depsAndProds"]').click().wait(1000);
```
Comment (aulonm) [inline tests/e2e/support/createProduct.js:6]: change to data-cy

File: tests/e2e/support/createTestDepartment.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+  cy.url().should('include', '/admin/data');
+
+  cy.get('[id="admin-addDepartment"]').click().wait(600);
+
+  cy.get('[id="dep-name"]').clear().type(testDepartment.name);
+
+  cy.get('#dep-missionStatement').clear().type(testDepartment.mission_statement);
+
+  cy.get('[id="btn-saveDep"]').click();
```
Comment (aulonm) [inline tests/e2e/support/createTestDepartment.js:16]: change id to data-cy

File: tests/e2e/support/deleteProduct.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import { testDepartment, testProducts } from '../config';
+
+Cypress.Commands.add('deleteProduct', productIndex => {
+  cy.visit('/admin').wait(2000);
+
+  cy.get('[id="admin-depsAndProds"]').click().wait(1000);
```
Comment (aulonm) [inline tests/e2e/support/deleteProduct.js:6]: change to data-cy

File: src/views/Admin/AdminHome.vue (hunk 2 above)
Comment (kiledal) [reply src/views/Admin/AdminHome.vue]: Fixed

File: src/mock/index.js (hunk 1 above)
Comment (kiledal) [reply src/mock/index.js]: requires the file inside a Promise() now which fixes this issue

**Pull request merged**